plugins/linguist-backend/README.md

```diff
@@ -112,6 +113,101 @@ return createRouter(
+
+As an example, you may choose to extend this to support both `Component` and `Resource` kinds along with allowing an opt-out annotation on the entity which entity authors can use.
+
+```ts
+new LinguistLanguageTagsProcessor({
+  logger: env.logger,
+  discovery: env.discovery,
+  shouldProcessEntity: (entity: Entity) => {
+    if (
+      ['Component', 'Resource'].includes(entity.kind) &&
+      !entity.metadata.annotations?.['backstage.io/ignore-linguist-tags']
+    ) {
+      return true;
+    }
+    return false;
+  },
+});
+```
+
+#### `tagTransformer`
+
+The `tagTransformer` option allows you to supply a custom function that can transform a linguist supplied language tag before it is appended to the entity tags. You can also use it to ignore certain language tags entirely by transforming the tag to `undefined`.
+
+The `defaultTagTransformer` is exported for your convenience to use in a chained transformer. It is recommended to still use it, as the default transformer will ensure that your language is valid per the [tags requirements](https://backstage.io/docs/features/software-catalog/descriptor-format/#tags-optional).
+
+As an example, one might use this supplied function to supply an allow list of languages.
+
+```ts
+new LinguistLanguageTagsProcessor({
+  logger: env.logger,
+  discovery: env.discovery,
+  tagTransformer: tag => {
+    const cleaned = defaultTagTransformer(lang);
+    return ['python', 'java', 'typescript'].includes(cleaned)
+      ? cleaned
+      : undefined;
+  },
+});
+```
+
+#### `cacheTTL`
+
+The `cacheTTL` option allows you to determine for how long this processor will cache languages for an `entityRef` before refreshing from the linguist backend. As this processor will run continuously, this cache is supplied to limit the load done on the linguist DB and API.
+
+By default, this processor will cache languages for 30 minutes before refreshing from the linguist database.
+
+You can optionally disable the cache entirely by passing in a `cacheTTL` duration of 0 minutes.
+
+```ts
+new LinguistLanguageTagsProcessor({
+  logger: env.logger,
+  discovery: env.discovery,
+  cacheTTL: { minutes: 0 },
+});
+```
+
+#### `bytesThreshold`
+
+The `bytesThreshold` option allows you to control a number of bytes threshold which must be surpassed before a language tag will be emitted by this processor. As an example, some repositories may have short build scripts written in Bash, but you may only want the main language of the project emitted (an alternate way to control this is to use the `tagTransformer` to reject any "bash" languages).
+
+```ts
+new LinguistLanguageTagsProcessor({
+  logger: env.logger,
+  discovery: env.discovery,
+  // Ignore languages with less than 5000 bytes in a repo.
+  bytesThreshold: 5000,
```

Comment:
It would be great to support options like that as configuration in `app-config.yaml` (e.g., also for `cacheTTL`)

Reply:
Do you mean instead or as an additional separate way to configure? I feel like splitting the config options between processor constructor args and config is not a great experience (there are two places to look at), and we do require processor constructor args for passing functions like `tagTransformer` and `shouldProcessEntity`.

Reply:
in general, I prefer a config-driven vs code-driven approach. You could still allow both if you like. I would also differentiate between the functions you can pass and the other configuration options. Those are rather extension points you provide.

In general, Backstage moves in the direction of declarative integration.

I guess, it would be good if you can add support for the new `@backstage/backend-plugin-api`. You can find examples if you search for `createBackendModule` (which would fit to your processor) and `createBackendPlugin`.

Not sure if this was done for the linguist-backend already or not.

Reply:
I've added support for `app-config.yaml`.